www/js/survey/enketo-survey-service.js

```diff
@@ -0,0 +1,137 @@
+  'emission.tripconfirm.service',
+])
+.factory('EnketoSurvey', function(
+  $window, $http, UnifiedDataLoader, Logger,
+  ConfirmHelper
+) {
+  var __form = null;
+  var __session = {};
+  var __form_location = null;
+  var __loaded_form = null;
+  var __loaded_model = null;
+
+  function init(formLocParam, optsParam) {
+    __form = null;
+    __session = {};
+    __form_location = formLocParam;
+    __loaded_form = null;
+    __loaded_model = null;
+
+    const opts = JSON.parse(optsParam);
+    if (opts && opts.session) {
+      __session = opts.session;
+    }
+
+    return $http.get(__form_location)
+    .then(function(form_json) {
+      __loaded_form = form_json.data.form;
+      __loaded_model = form_json.data.model;
+    });
+  }
+
+  function _loadForm(opts = {}) {
+    var formSelector = 'form.or:eq(0)';
+    var data = {
+      // required string of the default instance defined in the XForm
+      modelStr: __loaded_model,
+      // optional string of an existing instance to be edited
+      instanceStr: opts.instanceStr || null,
+      submitted: opts.submitted || false,
+      external: opts.external || [],
+      session: opts.session || {}
+    };
+    __form = new $window.FormModule( formSelector, data, {});
+    var loadErrors = __form.init();
+    return loadErrors;
+  }
+
+  function _restoreAnswer(answers) {
+    const answer = ConfirmHelper.getUserInputForTrip(__session.trip_properties, answers);
+    return (!answer) ? null : answer.data.survey_result;
+  }
+
+  function _parseAnswerByTagName(answerXml, tagName) {
+    const vals = answerXml.getElementsByTagName(tagName);
+    const val = vals.length ? vals[0].innerHTML : null;
+    if (!val) return '<null>';
+    return val.replace(/_/g, ' ');
+  }
+
+  function getAllSurveyAnswers(key = 'manual/confirm_survey', opts = {}) {
+    const _opts_populateLabels = opts.populateLabels || false;
+
+    const tq = $window.cordova.plugins.BEMUserCache.getAllTimeQuery();
+    return UnifiedDataLoader.getUnifiedMessagesForInterval(key, tq)
+    .then(function(answers){
+      if (!_opts_populateLabels) return answers;
+
+      const xmlParser = new $window.DOMParser();
+      if (key === 'manual/confirm_survey') {
+        return answers.map(function(answer){
+          const xmlStr = answer.data.survey_result;
+          const xml = xmlParser.parseFromString(xmlStr, 'text/xml');
+          // Data injection
+          answer.travel_mode_main = _parseAnswerByTagName(xml, 'travel_mode_main');
+          answer.o_purpose_main = _parseAnswerByTagName(xml, 'o_purpose_main');
+          answer.d_purpose_main = _parseAnswerByTagName(xml, 'd_purpose_main');
+          return answer;
+        });
+      }
+
+      return answers;
+    });
+  }
+
+  function displayForm() {
+    // Load survey with previous answer
+    if (__session &&
+        __session.data_key &&
+        __session.data_key === 'manual/confirm_survey'
+    ) {
+      return getAllSurveyAnswers(__session.data_key)
```

Comment:
@atton16, if you would access the `tripgj` object directly in this controller, you would be able to load the form with the survey results directly instead of making another call to retrieve them